pkg/timer/timerSync.go

```diff
@@ -61,14 +61,12 @@ func (ws *TimerSync) AddUpdateTimeTrigger(timeTrigger *fv1.TimeTrigger) {
 	ws.logger.Debug("cron event")
 
 	if item, ok := ws.timer.triggers[crd.CacheKeyUID(&timeTrigger.ObjectMeta)]; ok {
-		if item.trigger.Spec.Cron != timeTrigger.Spec.Cron {
-			if item.cron != nil {
-				item.cron.Stop()
-			}
-			item.trigger = *timeTrigger
-			item.cron = ws.timer.newCron(*timeTrigger)
-			logger.Debug("cron updated")
+		if item.cron != nil {
```

Comment:
We want to update cron only if there is any change in spec of time trigger. With these changes cron will get updated everytime even if there is no change in trigger spec.

You can add one more check in TimeTriggerEventHandlers function under updateFunc, something like
if oldtrigger.ObjectMeta.ResourceVersion != newtrigger.ObjectMeta.ResourceVersion {
					ws.AddUpdateTimeTrigger(timeTrigger)
				}